dataStructures/queues/queue.go

```diff
@@ -0,0 +1,80 @@
+/*
+Package main implements a priority queue data structure.
+
+Queues are commonly used for storing tasks that need to be done
+or incoming HTTP requests that need to be processed by a server.
+Handling interruptions in real-time systems, call handling,
+and CPU task scheduling are good examples for using queues.
+*/
+package main
+
+import (
+	"fmt"
+)
+
+// Queue data structure, it will be a list of Orders.
+type Queue []*Order
+
+// Order data type
+type Order struct {
+	priority     int
+	quantity     int
+	product      string
+	customerName string
+}
+
+// newOrder function contructs a new order.
+func newOrder(priority, quantity int, product, customerName string) *Order {
+	return &Order{
+		priority: priority,
+		quantity: quantity,
+		product: product,
+		customerName: customerName,
+	}
+}
+
+// Add method takes an order and adds it to the queue based on priority.
+func (q *Queue) Add(order *Order) {
+	if len(*q) == 0 {
+		*q = append(*q, order)
```

Comment:
Because we already know that q is empty, this can just be `*q = Queue{order}`

Reply:
I agree with you! Thanks.